File: backend/app/models/calendar.py
"""Pydantic models for calendar-related requests and responses."""

from pydantic import BaseModel
from typing import Optional


class CalendarEventResponse(BaseModel):
    id: str
    google_event_id: str
    title: str
    start_time: str
    end_time: str
    attendees: list = []
    description: Optional[str] = None
    meeting_id: Optional[str] = None
    synced_at: str


class CalendarSyncResponse(BaseModel):
    events_synced: int
    last_synced: str


class GoogleConnectionStatus(BaseModel):
    connected: bool
    email: Optional[str] = None
    last_synced: Optional[str] = None
    scopes: Optional[str] = None


class CalendarMatchSuggestion(BaseModel):
    calendar_event_id: str
    event_title: str
    event_date: str
    score: float
    match_reasons: list[str]


class LinkCalendarEventRequest(BaseModel):
    calendar_event_id: Optional[str] = None


class GoogleCallbackRequest(BaseModel):
    code: str

File: frontend/src/types/calendar.ts
export interface CalendarEvent {
  id: string;
  google_event_id: string;
  title: string;
  start_time: string;
  end_time: string;
  attendees: Array<{ email?: string; displayName?: string } | string>;
  description?: string;
  meeting_id?: string | null;
  synced_at: string;
}

export interface CalendarSyncResponse {
  events_synced: number;
  last_synced: string;
}

export interface GoogleConnectionStatus {
  connected: boolean;
  email?: string;
  last_synced?: string;
  scopes?: string;
}

export interface CalendarMatchSuggestion {
  calendar_event_id: string;
  event_title: string;
  event_date: string;
  score: number;
  match_reasons: string[];
}
